socorro/external/es/super_search_fields.py

```diff
@@ -20,6 +20,34 @@
 MAPPING_TEST_CRASH_NUMBER = 100
 
 
+def parse_mapping(mapping, namespace):
+    """Return set of fields in a mapping
+
+    This parses the mapping recursively.
+
+    :arg dict mapping: the mapping yielded by Elasticsearch
+    :arg str namespace: the namespace being parsed or None for the root
+
+    :returns: set of fields
+
+    """
+    fields = set()
+
+    for key in mapping:
+        field = mapping[key]
+        if namespace:
+            field_full_name = ".".join((namespace, key))
+        else:
+            field_full_name = key
+
+        if "properties" in field:
+            fields.update(parse_mapping(field["properties"], field_full_name))
+        else:
+            fields.add(field_full_name)
+
+    return fields
```

Comment:
This function was defined in the function that it was being used in. I extracted it to make it easier to use and test.